core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -486,56 +485,52 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
 
     Authority auth = Authority.fromString(uri.getAuthority());
     if (auth instanceof UnknownAuthority) {
-      // TODO(zac): In Alluxio 2.0 this warning will be upgraded to an exception
-      LOG.warn("Authority \"{}\" is unknown. The client will not be configured with this"
-          + " authority. The authority connection details will be loaded from your client"
-          + " configuration.",
-          auth);
-      mAlluxioHeader = getScheme() + ":///";
+      throw new IOException(String.format("Authority \"%s\" is unknown. The client can not be "
+              + "configured with the authority from %s", auth, uri));
     }
 
     mUri = URI.create(mAlluxioHeader);
-
     Map<String, Object> uriConfProperties = getConfigurationFromUri(uri);
 
-    synchronized (INIT_LOCK) {
-      if (sInitialized) {
-        if (!connectDetailsMatch(uriConfProperties, conf)) {
-          LOG.warn(ExceptionMessage.DIFFERENT_CONNECTION_DETAILS.getMessage(
-              FileSystemContext.get().getMasterInquireClient().getConnectDetails()));
-          initializeInternal(uriConfProperties, conf);
-        }
-      } else {
-        initializeInternal(uriConfProperties, conf);
-      }
-      // Must happen inside the lock so that the global filesystem context isn't changed by a
-      // concurrent call to initialize.
-      updateFileSystemAndContext();
-      sInitialized = true;
+    AlluxioProperties alluxioProps = ConfigurationUtils.defaults();
+    AlluxioConfiguration alluxioConf = mergeConfigurations(uriConfProperties, conf, alluxioProps);
```

Comment:
NIT: can we just inline `ConfigurationUtils.defaults()`, since we don't need to reuse that object again?

Reply:
done